addons/sale/models/sale_order_line.py

```diff
@@ -638,10 +638,13 @@ def _compute_product_packaging_id(self):
             # remove packaging if not match the product
             if line.product_packaging_id.product_id != line.product_id:
                 line.product_packaging_id = False
-            # Find biggest suitable packaging
+            # suggest biggest suitable packaging matching the SO's company
             if line.product_id and line.product_uom_qty and line.product_uom:
-                line.product_packaging_id = line.product_id.packaging_ids.filtered(
-                    'sales')._find_suitable_product_packaging(line.product_uom_qty, line.product_uom) or line.product_packaging_id
+                suggested_packaging = line.product_id.packaging_ids\
+                        .filtered('sales')\
+                        .filtered(lambda p: p.product_id.company_id <= p.company_id <= line.company_id)\
```

Comment:
```suggestion
                        .filtered(lambda p: p.product_id.company_id <= p.company_id <= line.company_id and p.sales)\
```
For performance reason. It will not be a huge problem here but in general we try to avoid extra loops

Reply:
Went with two `filtered` methods because it reads better, but that is a good point, updated it now.